packages/node/test/e2e/requester-fulfill.feature.ts

```diff
@@ -80,39 +80,38 @@ it('should call fail function on AirnodeRrp contract and emit FailedRequest if r
 
   await e2e.makeRequests(deployConfig, deployment);
 
-  const preinvokeLogs = await e2e.fetchAllLogs(provider, deployment.contracts.AirnodeRrp);
-
-  const preinvokeRegularRequests = preinvokeLogs.filter((log) => log.name === 'MadeTemplateRequest');
-  const preinvokeFullRequests = preinvokeLogs.filter((log) => log.name === 'MadeFullRequest');
-  const preinvokeFulfillments = preinvokeLogs.filter((log) => log.name === 'FulfilledRequest');
-
-  expect(preinvokeLogs.length).toEqual(5);
-  expect(preinvokeRegularRequests.length).toEqual(1);
-  expect(preinvokeFullRequests.length).toEqual(1);
-  expect(preinvokeFulfillments.length).toEqual(0);
+  const preInvokelogNames = (await e2e.fetchAllLogs(provider, deployment.contracts.AirnodeRrp)).map(({ name }) => name);
+  expect(preInvokelogNames).toEqual([
+    'SetAirnodeXpub',
+    'SetSponsorshipStatus',
+    'CreatedTemplate',
+    'MadeTemplateRequest',
+    'MadeFullRequest',
+  ]);
 
   const chain = e2e.buildChainConfig(deployment.contracts);
   const config = fixtures.buildConfig({ chains: [chain] });
   jest.spyOn(fs, 'readFileSync').mockReturnValue(JSON.stringify([config]));
 
   await handlers.startCoordinator();
 
-  const postinvokeLogs = await e2e.fetchAllLogs(provider, deployment.contracts.AirnodeRrp);
-
-  const postinvokeRegularRequests = postinvokeLogs.filter((log) => log.name === 'MadeTemplateRequest');
-  const postinvokeFullRequests = postinvokeLogs.filter((log) => log.name === 'MadeFullRequest');
-  const postinvokeFulfillmentsFailed = postinvokeLogs.filter((log) => log.name === 'FailedRequest');
-  const postinvokeFulfillments = postinvokeLogs.filter((log) => log.name === 'FulfilledRequest');
-
-  expect(postinvokeLogs.length).toEqual(7);
-  expect(postinvokeRegularRequests.length).toEqual(1);
-  expect(postinvokeFullRequests.length).toEqual(1);
-  expect(postinvokeFulfillmentsFailed.length).toEqual(1);
-  expect(postinvokeFulfillments.length).toEqual(1);
```

Comment:
If you look here these 2 expects are saying that one requests should have failed and the other one should have been fulfilled successfully.

I think that one fulfilled requests is missing in the expect on line 99 of the new code. On line 71 we are building a template requests that should run out of gas and then we create a full requests that should be fulfilled.

The problem that I saw when running this was that the second requests (which had a nonce of 1) was triggered before the first request (with nonce 0) causing the whole thing to fail.

Reply:
Sorry, I completely missed your comment from the other day :/

Anyway, I looked into this and the problem was when performing multiple sponsor requests. Each transaction has a nonce and the transactions need to be made sequentially (maybe can be batched by ethers provider, but I am not sure if that's a good idea).